File: GPSurb/GPSurb/Model/PriceModel.swift
//
//  PriceModel.swift
//  GPSurb
//
//  Created by Gilson Santos on 16/10/19.
//  Copyright © 2019 Gilson Santos. All rights reserved.
//

import Foundation

struct PriceModel: Codable {
    let currency: String?
    let currencyOriginal: String?
    let currentPrice: Double?
    let oldPrice: Double?
    let sku: String?
    let originalAmountPerDay: Double?
    let amountPerDay: Double?
    let amount: Double?
    
    enum CodingKeys: String, CodingKey {
        case currency = "currency"
        case currencyOriginal = "currency_original"
        case currentPrice = "current_price"
        case oldPrice = "old_price"
        case sku = "sku"
        case originalAmountPerDay = "originalAmountPerDay"
        case amountPerDay = "amountPerDay"
        case amount = "amount"
    }
}

File: app/src/main/java/viniciusapp/com/br/hurbtest/models/PriceModel.kt
package viniciusapp.com.br.hurbtest.models

import com.google.gson.annotations.SerializedName
import java.io.Serializable

data class PriceModel(
    @field:SerializedName("amount")
    var amount: Double? = null,
    @field:SerializedName("old_price")
    var old_price: Double? = null,
    @field:SerializedName("currency")
    var currency: String? = null,
    @field:SerializedName("currency_original")
    var currency_original: String? = null,
    @field:SerializedName("gain")
    var gain: Int? = null,
    @field:SerializedName("fee_extra_original")
    var fee_extra_original: Int? = null,
    @field:SerializedName("gain_original")
    var gain_original: Int? = null,
    @field:SerializedName("current_price")
    var current_price: Double? = null,
    @field:SerializedName("total_price")
    var total_price: Double? = null,
    @field:SerializedName("fee_extra")
    var fee_extra: Int? = null,
    @field:SerializedName("sku")
    var sku: String? = null,
    @field:SerializedName("taxes")
    var taxes: List<TaxesModel>? = null,
    @field:SerializedName("originalAmountPerDay")
    var originalAmountPerDay: Double? = null,
    @field:SerializedName("amountPerDay")
    var amountPerDay: Double? = null

) : Serializable